### PR TITLE
cafes: http registration

### PR DESCRIPTION
**Pull request by sanderpick**
Cafe registration over HTTP.

### CODE REVIEW

File: mobile/cafe.go [hunk 1]
```diff
@@ -6,12 +6,12 @@ import (
 )
 
 // RegisterCafe calls core RegisterCafe
-func (m *Mobile) RegisterCafe(peerId string) error {
-	if !m.node.Online() {
-		return core.ErrOffline
+func (m *Mobile) RegisterCafe(host string) error {
+	if !m.node.Started() {
```
Comment (sanderpick) [inline mobile/cafe.go]: Since we're not relying on the swarm, the node just needs to be started to register, not online.

File: core/api_cafes.go [hunk 2]
```diff
@@ -14,7 +14,7 @@ func (a *api) addCafes(g *gin.Context) {
 		return
 	}
 	if len(args) == 0 {
-		g.String(http.StatusBadRequest, "missing cafe id")
+		g.String(http.StatusBadRequest, "missing cafe host")
```
Comment (sanderpick) [inline core/api_cafes.go]: cli usage: `$ textile cafes add http://127.0.0.1:41601`

File: core/cafe_api.go [hunk 3]
```diff
@@ -24,12 +24,14 @@ import (
...
 )
 
+// CafeInfo details info about this cafe
 type CafeInfo struct {
 	Peer     string `json:"peer"`
 	Address  string `json:"address"`
 	API      string `json:"api"`
 	Protocol string `json:"protocol"`
 	Node     string `json:"node"`
+	URL      string `json:"url"`
```
Comment (sanderpick) [inline core/cafe_api.go]: The URL needed for registration is provided here. Useful when doing the gateway cafe discovery thing.

File: mobile/cafe.go (hunk 1 above)
Comment (asutula) [reply mobile/cafe.go]: cool, will be faster to launch the app the first time then

**Pull request merged**